### PR TITLE
Support SetVariable request

### PR DESCRIPTION
**Pull request by robertsipka**
Set the variable with the given name in the variable container to a new value

### CODE REVIEW

**Comment by robertsipka** [comment]
I've removed the WIP flag, ~~**PLEASE DO NOT MERGE**~~ until https://github.com/jerryscript-project/jerryscript/pull/2583 is not landed.

**Review by rtakacs — APPROVED** [review]
LGTM

**Comment by robertsipka** [comment]
The https://github.com/jerryscript-project/jerryscript/pull/2583 is landed, we could merge this PR after a second approve.

File: src/IotjsDebugger.ts [hunk 1]
```diff
@@ -501,6 +503,28 @@ class IotjsDebugSession extends DebugSession {
...
 
+  protected async setVariableRequest(response: DebugProtocol.SetVariableResponse,
+                                     args: DebugProtocol.SetVariableArguments
+  ): Promise <void> {
+    try {
+      const expression = args.name + '=' + args.value;
+      const scope_index = Number(this._variableHandles.get(args.variablesReference));
+      const result: JerryEvalResult = await this._protocolhandler.evaluate(expression, scope_index);
+      const value: string = result.subtype === EVAL_RESULT_SUBTYPE.JERRY_DEBUGGER_EVAL_OK
+      ? result.value
```
Comment (LaszloLango) [inline src/IotjsDebugger.ts]: This indentation looks odd to me. Is this the right style in this project?

File: src/test/JerryProtocolHandler.test.ts [hunk 2]
```diff
@@ -454,11 +454,18 @@ suite('JerryProtocolHandler', () => {
...
+      handler.evaluate('foo', 0);
       assert(debugClient.send.calledOnce);
+      const expression_length = [8, 0, 0, 0]; // length of chain index + eval subtype + length of code (4 + 1 + 3)
+      const scope_chain_index = [0, 0, 0, 0];
+      // chain index + eval subtype + code
+      const expression = [...scope_chain_index,
+                          SP.EVAL_SUBTYPE.JERRY_DEBUGGER_EVAL_EVAL.charCodeAt(0),
+                          'f'.charCodeAt(0), 'o'.charCodeAt(0), 'o'.charCodeAt(0)];
+
+
```
Comment (LaszloLango) [inline src/test/JerryProtocolHandler.test.ts]: I think one new line is enough.

File: src/JerryProtocolHandler.ts [hunk 3]
```diff
@@ -832,20 +832,31 @@ export class JerryDebugProtocolHandler {
...
     array[0] = SP.CLIENT.JERRY_DEBUGGER_EVAL;
     setUint32(this.byteConfig.littleEndian, array, 1, byteLength);
 
+    if (index < 0 || index > 65535) {
+      throw new Error('Invalid scope chain index');
+    }
+
+    setUint32(this.byteConfig.littleEndian, array, header_size, index);
+
+
```
Comment (LaszloLango) [inline src/JerryProtocolHandler.ts]: I think one new line is enough.

**Comment by robertsipka** [comment]
@LaszloLango : Thanks for the review, I've updated the PR.

**Review by LaszloLango — APPROVED** [review]
LGTM

**Pull request merged**